bredis/AsyncConnection.hpp

```diff
@@ -51,24 +53,75 @@ template <typename S> class AsyncConnection {
     std::mutex tx_queue_mutex_;
     std::mutex rx_queue_mutex_;
 
+    // I think that queuing should be the responsibility
+    // of a higher level interface and not this class.
+    // Instead, this class should just allow for one
+    // active asynchronous Redis read and one active
+    // asynchronous Redis write, with undefined behavior
+    // if two concurrent reads or two concurrent writes
+    // happen.
+    //
+    // Rationale: TCP/IP provides application-level flow
+    // control to match the rate of the receiver and the
+    // sender. In this implementation, the reader attempts
+    // to read as quickly as possible no matter how fast
+    // the consumer (in this case the command_callback_t)
+    // can process the data. This not only circumvents
+    // TCP/IP's natural application level flow control,
+    // but allows the size of the respective queues to
+    // grow without bounds. Furthermore, the caller is
+    // given no opportunity to provide "back pressure",
+    // nor is the caller able to detect the unbounded
+    // growth and impose its own policies on it.
+    // 
```

Comment:
I wasn't suggesting to disable Nagle's algorithm! You want Nagle's algorithm, in almost all cases it results in better performance. If you want to create an interface that allows multiple commands, consider this signature:
```
template<class FwdCommandRange, ...>
void
write_range(FwdCommandRange const& list, ...);
```

Where **FwdCommandRange** is a type that meets the requirements of **ForwardRange** (see http://www.boost.org/doc/libs/1_63_0/libs/range/doc/html/range/concepts/forward_range.html) and has `value_type` convertible to your `command_t` (or whatever its called).